doc/examples/mnist_mlp/mnist_loader.hpp

```diff
@@ -42,18 +42,18 @@ class MNistLoader
         return fread(loc, sizeof(T), n, m_file);
     }
 
-    std::uint32_t get_items() { return m_items; }
+    uint32_t get_items() { return m_items; }
 protected:
     std::string m_filename;
-    FILE* m_file{nullptr};
-    std::uint32_t m_magic;
-    std::uint32_t m_items;
+    FILE* m_file;
```

Comment:
Initializing with `{nullptr}` means every constructor gets the correct initialization if it's not explicit. Initialiing in the constructor means every constructor needs to explicitly initialize.

Reply:
I like consistency. There are some values that must be initialized in the ctor while others could be defined here in the header. If you want I can move all values that are not from parameters in the ctor to be initialized as they are defined.

Reply:
Let's start doing it that way; then the constructors only contain what is not the default.

Reply:
done